hackathon_site/dashboard/frontend/src/components/general/PopupModal/PopupModal.test.tsx

```diff
@@ -0,0 +1,21 @@
+import { render } from "../../../testing/utils";
+import PopupModal from "./PopupModal";
+import React from "react";
+
+test("renders general popup modal without crashing", () => {
+    const { getByText } = render(
+        <PopupModal
+            description={"Example Description"}
+            isVisible={true}
+            submitHandler={() => {}}
+            cancelHandler={() => {}}
+            cancelText={"cancel"}
+            submitText={"submit"}
+            title={"Test Modal"}
+        ></PopupModal>
+    );
+    expect(getByText("Example Description")).toBeInTheDocument();
+    expect(getByText("cancel")).toBeInTheDocument();
+    expect(getByText("submit")).toBeInTheDocument();
+    expect(getByText("Test Modal")).toBeInTheDocument();
```

Comment:
you need to test that the **cancel and submit functions** being passed into the component are being called.

e.g.
```
const submitFunction = jest.fn();
// render popup
fireEvent.click(getByText("submit"))
expect(submitFunction).toHaveBeenCalled()
```

Please do this in a separate unit test

Reply:
done!